Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -0,0 +1,214 @@
+                    }
+                    break;
+                case "IshObjectsGroup":
+                    if ((IshSession.ServerIshVersion.MajorVersion < 14) || ((IshSession.ServerIshVersion.MajorVersion == 14) && (IshSession.ServerIshVersion.RevisionVersion < 4)))
+                    {
+                        throw new PlatformNotSupportedException($"Add-IshBackgroundTask with the current parameter set requires server-side DocumentObj API which only available starting from 14SP4/14.0.4 and up. ServerIshVersion[{IshSession.ServerVersion}]");
+                    }
+                    break;
+            }
+            base.BeginProcessing();
+        }
+
+        /// <summary>
+        /// Process the cmdlet.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                if (IshObject != null)
+                {
+                    foreach (IshObject ishObject in IshObject)
+                    {
+                        _retrievedIshObjects.Add(ishObject);
+                    }
+                }
+            }
+            catch (TrisoftAutomationException trisoftAutomationException)
+            {
+                ThrowTerminatingError(new ErrorRecord(trisoftAutomationException, base.GetType().Name, ErrorCategory.InvalidOperation, null));
+            }
+            catch (Exception exception)
+            {
+                ThrowTerminatingError(new ErrorRecord(exception, base.GetType().Name, ErrorCategory.NotSpecified, null));
+            }
+        }
+
+        /// <summary>
+        /// Process the Add-IshBackgroundTask command-let.
+        /// </summary>
+        /// <exception cref="TrisoftAutomationException"></exception>
+        /// <exception cref="Exception"></exception>
+        /// <remarks>Writes <see cref="IshBackgroundTask"/> to the pipeline.</remarks>
+        protected override void EndProcessing()
+        {
+            try
+            {
+                IshFields metadataFilter = new IshFields();
+                IshFields requestedMetadata = IshSession.IshTypeFieldSetup.ToIshRequestedMetadataFields(IshSession.DefaultRequestedMetadata, ISHType, new IshFields(), Enumerations.ActionMode.Find);
+                long progressId = 0;
+                if (ParameterSetName == "IshObjectsGroup")
+                {
+                    var lngCardIds = _retrievedIshObjects.Select(ishObject => Convert.ToInt64(ishObject.ObjectRef[Enumerations.ReferenceType.Lng])).ToList();
+                    progressId = IshSession.DocumentObj25.RaiseEventByIshLngRefs(lngCardIds.ToArray(), EventType);
+                }
+
+                if (ParameterSetName == "ParameterGroup")
+                { 
+                    // Start event 
+                    var startEventRequest = new StartEventRequest
+                    {
+                        description = EventDescription,
```

Comment:
EventDescription is not mandatory (yet)... so what happens in Web Client if you pass empty string?

Reply:
For now I thought we made the parameter mandatory, but if you would pass an empty string then the description will become the same as the eventType

Reply:
Set EventDescription to mandatory and validating only for a null "ValidateNotNull"

Reply:
Done